test_container/tests/test/standard-flavor/all/import_modules.py

```diff
@@ -0,0 +1,203 @@
+                    self.modules = []
+
+                def module_limit_reached(self) -> bool:
+                    self.module_count += 1                                                                                          
+                    return self.module_count > self.module_limit
+                
+                def error_limit_reached(self) -> bool:
+                    self.error_count += 1                                                                                          
+                    return self.error_count > self.error_limit
+
+                def import_modules(self):
+                    self.modules = get_module_names(sys.path)
+                    while len(self.modules) > 0:
+                        module = self.modules.pop()
+                        limit_reached = self.import_module(module)
+                        if limit_reached:
+                            break
+
+                def add_submodules(self, module, module_import):
+                    if hasattr(module_import,"__path__"):
+                        submodule_names = get_module_names(module_import.__path__)
+                        self.modules.extend([
+                            f"{module}.{submodule}"
+                            for submodule in submodule_names 
+                            if not submodule.startswith("_") and submodule not in self.excluded_submodules
+                        ])
+
+                def import_module(self, module: str) -> bool:
+                    print("========================================================")
+                    print("========================================================")
+                    print("========================================================")
+                    print("modules left: ", len(self.modules))
+                    print("current module: ", module)
+                    print("========================================================")
+                    print("========================================================")
+                    print("========================================================", flush=True)
+                    if module in self.excluded_modules or module.startswith("_"):
+                        self.ctx.emit(module, None, "SKIPPED")
+                        return self.module_limit_reached()
+                    try:
+                        module_import = importlib.import_module(module)
+                        self.ctx.emit(module, None, "OK")
+                        self.add_submodules(module=module, module_import=module_import)
+                        return self.module_limit_reached()
+                    except BaseException as e:
+                        import traceback
+                        if hasattr(e,"msg") and e.msg == "No module named 'pytest'":
+                            self.ctx.emit(module, None, "IGNORED")
+                            return self.module_limit_reached()
```

Comment:
If a module can't be imported because it depends on 'pytest', shouldn't it be counted as error? Because, I mean, if customers want to use that module, they will run into the same error, right?

Reply:
Background all modules that import pytest, are actually tests that are shipped with package  We can remove this check, but then our exclude lists probably get longer.